package.py

```diff
@@ -988,6 +995,23 @@ def build(self) -> None:
         else:
             self.__move_to_dist()
 
+    @log_group('cargo install')
+    def __cargo_build(self) -> None:
+        colibri_directory = self.__storage.colibri_directory
+        build_ret_code = subprocess.call(
+            f'cargo build --target-dir {colibri_directory} --manifest-path ./colibri/Cargo.toml --release',
+            shell=True,
+        )
+        if build_ret_code != 0:
+            logger.error('packaging failed')
+            sys.exit(1)
+
+        backend_binary = colibri_directory / 'release' / 'colibri'
+        ret_code = subprocess.call(f'{backend_binary}', shell=True)
+        if ret_code != 0:
+            logger.error('colibri binary check failed')
+            sys.exit(1)
```

Comment:
For macOS we need to build for two different targets and then merge with lipo

`aarch64-apple-darwin` and `x86_64-apple-darwin`

for the merging, I think we can use this

https://github.com/rotki/rotki/blob/ea75e4aeec8d79d0b5e2e8463ddb9666fc095f1a/package.py#L430

I am not sure how we can use `rustup` from the action to support both the architectures targets